cpp/unit/ability.h

```diff
@@ -42,17 +43,14 @@ using ability_set = std::bitset<16>;
 using ability_id_t = unsigned int;
 
 /**
- * all bits set to 1
+ * set bits corresponding to abilities
  */
-const ability_set ability_all = ability_set().set();
+ability_set from_list(const std::vector<ability_type> &items);
```

Comment:
yeah, but `from_list` still pollutes the global namespace, and its name doesn't indicate in any way that it's specific to ability sets.